### PR TITLE
Fix races

### PR DESCRIPTION
**Pull request by emilevauge**
This PR fixes some races in Traefik.
The Docker provider still needs to be fixed /cc @vdemeester

### CODE REVIEW

File: safe/routine.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	"sync"
+)
+
+type routine struct {
+	goroutine func(chan bool)
+	stop      chan bool
+}
+
+// Pool creates a pool of go routines
+type Pool struct {
```
Comment (vdemeester) [inline safe/routine.go]: What about using `sync.Pool` here ? :angel:

Comment (emilevauge) [inline safe/routine.go]: Because I don't want these objects to be GC ;)

Comment (vdemeester) [inline safe/routine.go]: hum

**Comment by vdemeester** [comment]
LGTM 🐯

**Pull request merged**